cmd/compose/compose.go

```diff
@@ -206,6 +206,19 @@ func (o *projectOptions) toProject(services []string, po ...cli.ProjectOptionsFn
 
 	project.WithoutUnnecessaryResources()
 
+	for _, s := range project.Services {
+		for i, v := range s.Volumes {
+			if projectVolume, ok := project.Volumes[v.Source]; ok &&
+				projectVolume.DriverOpts["o"] == types.VolumeTypeBind {
+				s.Volumes[i].Source = project.Name + "_" + s.Volumes[i].Source
```

Comment:
I'm not a big fan we hack the model this way. Maybe the "use bind vs mount" logic used in `create` could check the volume driver and use bind when necessary?

Reply:
PTAL